Library/Homebrew/compilers.rb

```diff
@@ -45,7 +48,7 @@ def self.create(spec, &block)
 
   def initialize(name, version, &block)
     @name = name
-    @version = version
+    @version = Version.new(version.to_s)
```

Comment:
Maybe use `Version.parse` on the stringified value? That'll assign `Version::NULL` as appropriate.